File: src/main/java/dev/tuhkanens/comfortlib/api/UpdateAPI.java
package dev.tuhkanens.comfortlib.api;

import dev.tuhkanens.comfortlib.result.UpdateResult;

public interface UpdateAPI {
    void setEnabled(boolean enabled);
    void setProjectId(String projectId);
    void setVersion(String version);

    boolean getEnabled();
    String getProjectId();
    String getVersion();

    UpdateResult checkUpdates();
}

File: src/main/kotlin/dev/tuhkanens/comfortlib/api/UpdateAPI.kt
package dev.tuhkanens.comfortlib.api

import dev.tuhkanens.comfortlib.update.UpdateCheckResult

interface UpdateAPI {
    fun setEnabled(enabled: Boolean)
    fun setProjectId(projectId: String)
    fun setVersion(version: String)

    fun getEnabled(): Boolean
    fun getProjectId(): String
    fun getVersion(): String

    fun checkUpdates(): UpdateCheckResult
}